modules/cloudflarepurger/src/Plugin/Purge/Purger/CloudFlarePurger.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\cloudflare\Plugin\PurgePurger\CloudFlarePurger.
+ */
+
+namespace Drupal\cloudflarepurger\Plugin\Purge\Purger;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\cloudflare\CloudFlareStateInterface;
+use Drupal\cloudflarepurger\EventSubscriber\CloudFlareCacheTagHeaderGenerator;
+use Drupal\purge\Plugin\Purge\Purger\PurgerBase;
+use Drupal\purge\Plugin\Purge\Purger\PurgerInterface;
+use Drupal\purge\Plugin\Purge\Invalidation\InvalidationInterface;
+use CloudFlarePhpSdk\ApiEndpoints\CloudFlareAPI;
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * CloudFlare purger.
+ *
+ * @PurgePurger(
+ *   id = "cloudflare",
+ *   label = @Translation("CloudFlare"),
+ *   description = @Translation("Purger for CloudFlare."),
+ *   configform = "Drupal\cloudflare\Form\CloudFlareAdminSettingsForm",
+ *   types = {"tag", "url", "everything"},
+ *   multi_instance = FALSE,
+ * )
+ */
+class CloudFlarePurger extends PurgerBase implements PurgerInterface {
+
+  /**
+   * The settings configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
```

Comment:
As said before, I wan't logging out of purgers and ``purge `` to take control over it. The most recent "queuers and processors" merge unblocks me and it could be this week that this lands (fingers crossed). Then, in purge's most verbose logging setting, it will log something like ``purger cloudflare got offered 12 objects of types tag, url`` and ``purger cloudflare has marked all objects as succeeded``. 

Feel free to keep your own logging at purger level, just know that centrally organized logging is coming.